curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-debugging-by-building-a-random-background-color-changer/66326913b729e16dd0127a36.md

```diff
@@ -0,0 +1,237 @@
+---
+id: 66326913b729e16dd0127a36
+title: Step 7
+challengeType: 0
+dashedName: step-7
+---
+
+# --description--
+
+CamperBot has finished building out their random background color changer. However, when they click the button, the background color does not change. 
+
+It looks like they are trying to use the `onclick` property but they are using it incorrectly.  The `onclick` property should be assigned a function reference.
+
+Fix the error in the `btn.onclick = changeBackgroundColor();` line.
+
+Remember that you worked with the `onclick` property in the Role playing game project. Look back at the final solution to see how `onclick` was properly used.
+
+Once you fix that final bug, the random background color changer will be complete!
+
+# --hints--
+
+You should not call the `changeBackgroundColor` function. You should instead assign the function reference to the `onclick` property.
+
+```js
+assert.strictEqual(btn.onclick, changeBackgroundColor);
```

Comment:
Not necessarily blocking, buuuuuut this means that the valid code won't pass:

```js
btn.onclick = (() => changeBackgroundColor());
```

Not sure if we want to account for that.

Reply:
I am having a hard time get both solutions to pass with the updated tests
Each additional test I try, I can only get the `btn.onclick = changeBackgroundColor` to pass but not the other solution.